src/components/ApiReference/ScrollRouter.js

```diff
@@ -34,10 +37,14 @@ export const ScrollRouter = ({ children, initialActive = "" }) => {
   const isScrollingDown = React.useRef(false);
 
   // Navigation
-  const onLinkClick = React.useCallback(function onLinkClick(route) {
-    window.history.pushState(null, null, route);
-    smoothScrollTo(elementMap.get(route).current, { duration: 0 });
-  }, []);
+  const onLinkClick = React.useCallback(
+    function onLinkClick(route) {
+      emit(route);
```

Comment:
There's no mechanism to watch for URL changes on a page level, so I updated the scroll router to support subscriptions. Any time the route updates, all subscribers will be notified.